spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -64,6 +65,7 @@
 	private volatile EvaluationContext evaluationContext;
 
 	@Autowired(required=false) // this would need to be refactored into constructor in the future
+	@StreamRetryTemplate
 	private RetryTemplate retryTemplate;
```

Comment:
This technique correctly fails to load with the following:

```java

	@Bean
	@StreamRetryTemplate
	public RetryTemplate retryTemplate1() {
		return new RetryTemplate();
	}

	@Bean
	@StreamRetryTemplate
	public RetryTemplate retryTemplate2() {
		return new RetryTemplate();
	}

```
```
Field retryTemplate in org.springframework.cloud.stream.binder.AbstractBinder required a single bean, but 2 were found:
	- retryTemplate1: defined by method 'retryTemplate1' in org.springframework.cloud.stream.binder.test.SampleStreamApp
	- retryTemplate2: defined by method 'retryTemplate2' in org.springframework.cloud.stream.binder.test.SampleStreamApp
```

However, with this...

```java
	@Bean
	@StreamRetryTemplate
	public RetryTemplate retryTemplate() {
		return new RetryTemplate();
	}

	@Bean
	@StreamRetryTemplate
	public RetryTemplate retryTemplate2() {
		return new RetryTemplate();
	}

```
The context loads ok (the first template is injected - by name).

Maybe we should rename the field to something like `consumerBindingRetryTemplate` ?

Then we'll get

```
Field consumerBindingRetryTemplate in org.springframework.cloud.stream.binder.AbstractBinder required a single bean, but 2 were found:
	- retryTemplate: defined by method 'retryTemplate' in org.springframework.cloud.stream.binder.test.SampleStreamApp
	- retryTemplate2: defined by method 'retryTemplate2' in org.springframework.cloud.stream.binder.test.SampleStreamApp
```

Or give it some obscure name so a user is unlikely to configure a bean with that name.

Reply:
Hmm. . . let me think a bit